sdk/personalizer/Azure.AI.Personalizer/src/Models/PersonalizerClient.cs

```diff
@@ -151,6 +156,8 @@ public virtual async Task<Response<PersonalizerRankResult>> RankAsync(Personaliz
                 }
                 else
                 {
+                    Models.PersonalizerClientProperties result = await ClientConfigurationRestClient.PostAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Calling the ClientConfigurationClient here instead when isLocalInference is true (these two lines will be moved)  just to test the new code is working when RankTest is called. But few things have to be modified while testing for the test case to work. Tested in my local machine and it worked as below.
Standard Output: 
InternalId: 86aa7660871843dca1c703400557b71c
StorageBlobUri: BlobURI
LearningMode: 0
Exploration Percentage: 1
initialCommandline: --cb_explore_adf --json --quiet --epsilon 0.0 --first_only --id N/A

Reply:
The point of the thick client, is decrease latency. By making a configuration network request on every rank call, the latency is effectively increased. This configuration should also be shared across the entire client, not just the rank call. Consider moving to a wider scope and changing the client configuration to a TTL lazy initialization.